.devops/nix/package.nix

```diff
@@ -114,14 +114,22 @@ effectiveStdenv.mkDerivation (
     pname = "llama-cpp${pnameSuffix}";
     version = llamaVersion;
 
+    # Note: none of the fiels that are filtered out are visible in the sandbox
+    # or affect the output hash. This also means they can be modified without
+    # triggering a rebuild.
     src = lib.cleanSourceWith {
       filter =
         name: type:
-        !(builtins.any (_: _) [
+        let
+          noneOf = builtins.all (x: !x);
+          baseName = baseNameOf name;
+        in
+        noneOf [
           (lib.hasSuffix ".nix" name) # Ignore *.nix files when computing outPaths
-          (name == "README.md") # Ignore *.md changes whe computing outPaths
-          (lib.hasPrefix "." name) # Skip hidden files and directories
-        ]);
+          (lib.hasSuffix ".md" name) # Ignore *.md changes whe computing outPaths
+          (lib.hasPrefix "." baseName) # Skip hidden files and directories
+          (baseName == "flake.lock")
+        ];
```

Comment:
Could also make this a white-list like https://github.com/ggerganov/llama.cpp/blob/0ea069b87bd296c556824e57455433b6c0357340/.github/workflows/nix-ci.yml#L8-L11.

Then a successful build would constitute a "proof" that no other paths are required or affect the output (the c++ portion of the program)

Reply:
I'm not particularly fussed by extra rebuilds on a fast-moving project like this. If we skipped this, and rebuilt on every change, it's not so bad, in my view. Nice, but a source of extra complexity that could skip.